File: 0547-number-of-provinces/0547-number-of-provinces.py
from collections import deque
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        
        def bfs(isConnected,visited,s):
            
            queue=deque()
            queue.append(s)
            visited[s]=True
            
            while queue:
                
                curr=queue.popleft()
                
                for neighbour in range(V):
                    if isConnected[curr][neighbour]==1 and not visited[neighbour]:
                        queue.append(neighbour)
                        visited[neighbour]=True
        V=len(isConnected)
        visited=[False]*V
        count=0
        for i in range(V):
            if not visited[i]:
                count+=1   
                bfs(isConnected,visited,i)
        return count

File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
private:
    void bfs(int src,vector<vector<int>>& isConnected,vector<int> &vis,int V){
        queue<int> q;
        vis[src]=1;
        q.push(src);
        
        while(!q.empty()){
            
            int curr=q.front();
            q.pop();
            for(int n=0;n<V;n++){
                if(isConnected[curr][n]==1 && !vis[n]){
                    q.push(n);
                    vis[n]=1;
                }
            }
        }
    }
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        int V=isConnected.size();
        vector<int> vis(V,0);
        int count=0;
        for(int i=0;i<V;i++){
            if(!vis[i]){
                count+=1;
                bfs(i,isConnected,vis,V);
            }
        }
        return count;
    }
};